applications/nrf5340_audio/broadcast_sink/main.c

```diff
@@ -303,6 +309,99 @@ static void le_audio_msg_sub_thread(void)
 	}
 }
 
+/**
+ * @brief	Handle bt_mgmt events.
+ */
+static void bt_mgmt_msg_sub_thread(void)
+{
+	int ret;
+	static uint8_t broadcast_code[BT_AUDIO_BROADCAST_CODE_SIZE];
+	const struct zbus_channel *chan;
+
+	while (1) {
+		struct bt_mgmt_msg msg;
+
+		ret = zbus_sub_wait_msg(&bt_mgmt_evt_sub, &chan, &msg, K_FOREVER);
+		ERR_CHK(ret);
+
+		switch (msg.event) {
+		case BT_MGMT_CONNECTED:
+			LOG_DBG("Connected");
+			break;
+
+		case BT_MGMT_DISCONNECTED:
+			LOG_DBG("Disconnected");
+			break;
+
+		case BT_MGMT_SECURITY_CHANGED:
+			LOG_DBG("Security changed");
+			break;
+
+		case BT_MGMT_PA_SYNCED:
+			LOG_DBG("PA synced");
+
+			ret = broadcast_sink_pa_sync_set(msg.pa_sync, msg.broadcast_id);
+			if (ret) {
+				last_broadcast_id = BRDCAST_ID_NOT_USED;
+				LOG_WRN("Failed to set PA sync");
+			} else {
+				last_broadcast_id = msg.broadcast_id;
+			}
+
+			break;
+
+		case BT_MGMT_PA_SYNC_LOST:
+			LOG_INF("PA sync lost, reason: %d", msg.pa_sync_term_reason);
+
+			if (IS_ENABLED(CONFIG_BT_OBSERVER) &&
+			    msg.pa_sync_term_reason != BT_HCI_ERR_LOCALHOST_TERM_CONN) {
+				ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_BROADCAST, NULL,
+							 BRDCAST_ID_NOT_USED);
+				if (ret) {
+					if (ret == -EALREADY) {
+						return;
+					}
+
+					LOG_ERR("Failed to restart scanning: %d", ret);
+					break;
```

Comment:
Should not have a return in this thread, it will kill the thread.
```suggestion
					if (ret != -EALREADY) {
						LOG_ERR("Failed to restart scanning: %d", ret);
					}
					
					break;
```